openapi/platform/platform.yaml

```diff
@@ -1802,7 +1802,79 @@ paths:
           required: true
       security:
         - OAuth2:
-            - automation:read
+            - 'automation:read'
+      tags:
+        - Automations
+  /automations:
+    parameters: []
+    get:
+      summary: List Automations
+      responses:
+        '200':
+          description: OK
+          content:
+            application/vnd.api+json:
+              schema:
+                type: array
+                description: ''
+                minItems: 1
+                uniqueItems: true
+                x-examples:
+                  example-1:
+                    - type: automations
+                      id: Automation-91cb048c-c498-43ba-ab99-2ca2d27d2c46
+                      attributes:
+                        name: Example Automation
+                items:
+                  type: object
+                  properties:
+                    type:
+                      type: string
+                      minLength: 1
+                    id:
+                      type: string
+                      minLength: 1
+                    attributes:
+                      type: object
+                      required:
+                        - name
+                      properties:
+                        name:
+                          type: string
+                          minLength: 1
+                  required:
+                    - type
+                    - id
+                    - attributes
+              examples:
+                example-1:
+                  value:
+                    data:
+                      - id: Automation-abcd1234-12ab-12ab-12ab-12ab12ab12ab
+                        type: automations
+                        attributes:
+                          name: Example Automation
+        '':
+          description: ''
+      operationId: list-automations
+      x-lifecycle:
+        status: trustedTester
+      description: |-
+        [Status](https://developers.vendasta.com/platform/ZG9jOjEwMTU2NTYy-versioning): `Trusted Tester`
+
+        Returns a list of automations.
+      parameters:
```

Comment:
You will need to document the query parameters as well.

`filter[businessPartner.id]` - Return automations defined for the specified partner

`filter[name]` - Filter by partial name match, case insensitive (?)

`filter[status]` - Only return automations with the specified state(s). Valid values are: `on`, 
`off`, `draining`, `any`

`filter[tags]` - Only return automations with all of the specified tags. You may either provide a comma-separated list of tags or specify the parameter multiple times.

`filter[trigger]` - I'm not sure how this one should work. For Zapier I need a way to filter to `TaskDefinition-manual-api-order-trigger` or `TaskDefinition-manual-api-account-trigger`. It might make sense to let partners filter for other trigger types as well. 


You will also need
`page[cursor]` - The cursor stores all your filters and current location in the list to allow paging over the results in smaller batches. The value will be provided in the response links.

`page[limit]` - The maximum number of automations you would like returned in a single batch. Use the links.next member in the response to get the remainder.

Reply:
Added in most of these- I'm not sure about the trigger filter either right now. Our automata List endpoint doesn't currently support this so I'm going to add support there first to better understand how this might look here

Reply:
The frontend currently calls `/automata.v1.Automata/ListAutomations` with a filter of `"triggerTaskDefinitionId":"TaskDefinition-manual-api-account-trigger"`. That is what I am looking for with the `filter[trigger]`. (Multi-select would be nice but is not needed as that is not currently supported). 

Would we want to treat it as an attribute with an enum of valid values or a relationship to a future list triggers endpoint?

Reply:
Oh yup you're right, my postman collection must be out of date since it didn't have that filter.

> Would we want to treat it as an attribute with an enum of valid values or a relationship to a future list triggers endpoint?

That's a good question- I think for people using this endpoint an enum would cover the practical use cases, be the most prescriptive, & leave room for us to switch to strings if we need. We really only need to update the enum when we add a new api trigger type, which is rare.

Reply:
I agree with you. I can't think of a good reason that partners would want to get a list of automations for other trigger types. Eventually, we will let vendors define their own triggers but I don't think that will tie into this API.